network/transport/tcp_socket.c

```diff
@@ -70,20 +108,23 @@ struct tcp_socket_desc {
  *  - \ref SUCCESS : On success
  *  - \ref FAILURE : Otherwise
  */
-int32_t socket_init(struct tcp_socket_desc **desc,
-		    struct tcp_socket_init_param *param)
+static int32_t tcp_socket_init(struct tcp_socket_desc **desc,
+			       struct network_interface *net,
+			       struct tcp_socket_init_param *param)
 {
 	struct tcp_socket_desc	*ldesc;
 	int32_t			ret;
 
-	if (!desc || !param)
+	UNUSED_PARAM(param);
```

Comment:
i think this is unused and also it is an empty structure. It looks redundant to have it like this. Maybe just to have a socket_init_param for both tcp and tls would make more sense and just not use it for tcp if you do not need anything from it. To have an empty structure is useless.

Reply:
Explained here https://github.com/analogdevicesinc/no-OS/pull/581#discussion_r423653253